plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/element/ShutDownVpcVspCommand.java

```diff
@@ -40,9 +43,61 @@ public String getVpcUuid() {
         return _vpcUuid;
     }
 
+    public String getDomainTemplateName() {
+        return _domainTemplateName;
+    }
+
+    public static class Builder implements CmdBuilder<ShutDownVpcVspCommand> {
+        private String _domainUuid;
+        private String _vpcUuid;
+        private String _domainTemplateName;
+
+        public Builder domainUuid(String domainUuid) {
+            this._domainUuid = domainUuid;
+            return this;
+        }
+
+        public Builder vpcUuid(String vpcUuid) {
+            this._vpcUuid = vpcUuid;
+            return this;
+        }
+
+        public Builder domainTemplateName(String domainTemplateName) {
+            this._domainTemplateName = domainTemplateName;
+            return this;
+        }
+
+        @Override
+        public ShutDownVpcVspCommand build() {
+            return new ShutDownVpcVspCommand(_domainUuid, _vpcUuid, _domainTemplateName);
+        }
+    }
+
     @Override
     public boolean executeInSequence() {
         return false;
     }
 
+    @Override
+    public boolean equals(Object o) {
```

Comment:
Implemented <code>equals</code> method for the <code>Command</code> super class, calling it from subclass.